src/js/player.js

```diff
@@ -1659,6 +1662,37 @@ vjs.Player.prototype.isAudio = function(bool) {
   return this.isAudio_;
 };
 
+/**
+ * Trigger an event on the player.
+ *
+ *     myPlayer.trigger('eventName');
+ *     myPlayer.trigger({'type':'eventName'});
+ *
+ * Event names will be prefixed with the current value of this.eventPrefix()
+ *
+ * @param  {Event|Object|String} event  A string (the type) or an event object with a type attribute
+ * @return {vjs.Player}       self
+ */
+vjs.Player.prototype.trigger = function(event) {
+  if (this.eventPrefix_ !== '') {
+    if (typeof event === 'string') {
+      event = this.eventPrefix_ + event;
+    } else if (typeof event === 'object') {
+      event.type = this.eventPrefix_ + event.type;
+    }
+  }
+  return vjs.Component.prototype.trigger.apply(this, arguments);
+};
+vjs.Player.prototype.eventPrefix = function(prefix) {
```

Comment:
can we add documentation to strongly discourage the use of this method by regular users?